src/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs

```diff
@@ -1393,10 +1419,18 @@ private void Indent(bool beforeEndElement)
             else if (!_stack[_top].mixed)
             {
                 _textWriter.WriteLine();
-                int i = beforeEndElement ? _top - 1 : _top;
-                for (i *= _indentation; i > 0; i--)
+                int i = (beforeEndElement ? _top - 1 : _top) * _indentation;
+                if(i <= _indentChars.Length)
```

Comment:
The typical case is going to evaluate that as true. In the while loop there's a call to Math.Min, an extra call to _indentChars.Length and a decrement of i which doesn't exist in the base case. When i is small, this will add a significant amount of overhead and this is very hot code.